src/msg/async/EventKqueue.cc

```
@@ -199,13 +199,14 @@ int KqueueDriver::resize_events(int newsize)
   ldout(cct,30) << __func__ << " kqfd = " << kqfd << "newsize = " << newsize 
                 << dendl;
   if (newsize > sav_max) {
-    sav_events = (struct SaveEvent*)realloc(sav_events, sizeof(struct SaveEvent)*newsize);
-    if (!sav_events) {
+    struct SaveEvent* new_events = (struct SaveEvent*)realloc(sav_events, sizeof(struct SaveEvent)*newsize);
+    if (!new_events) {
       lderr(cct) << __func__ << " unable to realloc memory: "
                              << cpp_strerror(errno) << dendl;
-      ceph_assert(sav_events);
+      ceph_assert(new_events);
```

@wanglinke521 thanks for the fix. we had a series of discussion around this piece of code before, see #15088 and #21488. could you please take a look at them ? the take away is that we don't have proper error handling, so no need to pretend that we have proper recovery here, and better off fail early.  probably a better fix is to replace `ceph_assert()` with `ceph_abort()`. what do you think?

---

Thank you for your reply. I read your previous discussion and understood the ins and outs of this problem. I also think that ceph_abort seems to be more appropriate here. Because sav_events is null, it is determined to have failed, so there is no need to assert it again.